scr/main/java/bittercode/util/StoreUtil.java

```diff
@@ -0,0 +1,86 @@
+package com.bittercode.util;
+
+import java.io.PrintWriter;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
+import com.bittercode.model.UserRole;
+
+/*
+ * Store UTil File To Store Commonly used methods
+ */
+public class StoreUtil {
+
+    /**
+     * Check if the User is logged in with the requested role
+     */
+    public static boolean isLoggedIn(UserRole role, HttpSession session) {
+
+        return session.getAttribute(role.toString()) != null;
+    }
+
+    /**
+     * Modify the active tab in the page menu bar
+     */
+    public static void setActiveTab(PrintWriter pw, String activeTab) {
+
+        pw.println("<script>document.getElementById(activeTab).classList.remove(\"active\");activeTab=" + activeTab
+                + "</script>");
+        pw.println("<script>document.getElementById('" + activeTab + "').classList.add(\"active\");</script>");
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/2#issuecomment-2418799949>#b11fc3</a><span> - 10/17/2024, 08:03 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The setActiveTab method directly inserts the activeTab parameter into a script tag without any sanitization. This could potentially lead to XSS vulnerabilities if the activeTab value is user-controlled. Consider using a proper JavaScript escaping function or a template engine that automatically escapes output.
</div>
<div id="fix">
<b>Fix:</b> Implement proper output encoding for the activeTab parameter before inserting it into the script tag. Use a well-tested library or framework function for JavaScript string escaping.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -26,6 +26,7 @@
  public static void setActiveTab(PrintWriter pw, String activeTab) {

 +        String escapedActiveTab = StringEscapeUtils.escapeEcmaScript(activeTab);
 -        pw.println("<script>document.getElementById(activeTab).classList.remove(\"active\");activeTab=" + activeTab
 -                + "</script>");
 -        pw.println("<script>document.getElementById('" + activeTab + "').classList.add(\"active\");</script>");
 +        pw.println("<script>document.getElementById(activeTab).classList.remove(\"active\");activeTab='" + escapedActiveTab + "';</script>");
 +        pw.println("<script>document.getElementById('" + escapedActiveTab + "').classList.add(\"active\");</script>");

  }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged